src/main/kotlin/com/github/devcordde/devcordbot/core/autohelp/Conversation.kt

```diff
@@ -0,0 +1,210 @@
+ * @property helpMessage the helpmessage sent by the bot
+ * @property lastInteraction the [OffsetDateTime] of the last interaction in this conversation
+ * @property answer the [ConversationAnswer]
+ */
+data class Conversation(
+    val owner: Member,
+    val trigger: Message,
+    val stacktraces: MutableList<StackTrace>,
+    val classes: MutableList<Class>,
+    var helpMessage: Message?,
+    var lastInteraction: OffsetDateTime,
+    val answer: ConversationAnswer = ConversationAnswer()
+) {
+    /**
+     * The text Channel the message is in.
+     */
+    val textChannel: TextChannel
+        get() = trigger.textChannel
+
+    /**
+     * Creates message if needed.
+     */
+    private val safeHelpMessage: Message
+        get() {
+            if (helpMessage == null) {
+                helpMessage = trigger.channel.sendMessage(
+                    Embeds.loading(
+                        "Ich denke nach!",
+                        "Ich versuche zu helfen bitte warte etwas."
+                    )
+                ).complete()
+            }
+            return helpMessage!!
+        }
+
+    /**
+     * Updates the Discord answer message.
+     */
+    fun update(): Unit =
+        safeHelpMessage.editMessage(answer.toEmbed()).queue()
+}
+
+/**
+ * Mutable object to store information on how to answer an error.
+ *
+ * @property exception the exception daat
+ * @property causeContent the raw string of the line causing the issue
+ * @property useless whether this answer does contain any useful info or not
+ * @property isComplete whether all information was found or not
+ *  @property npeHint hint to resolve an NPE
+ */
+class ConversationAnswer {
+    lateinit var exception: ExceptionAnswer
+    lateinit var causeContent: String
+    lateinit var npeHint: String
+
+    val useless: Boolean
+        get() = !::exception.isInitialized && !::causeContent.isInitialized
+
+    val isComplete: Boolean
+        get() = ::exception.isInitialized && ::causeContent.isInitialized
+
+    internal val exceptionSet: Boolean
+        get() = ::exception.isInitialized
+
+    internal val causeSet: Boolean
+        get() = ::causeContent.isInitialized
+
+    internal val npeHintSet: Boolean
+        get() = ::npeHint.isInitialized
+
+    /**
+     * Info about the exception.
+     *
+     * @property exceptionName the name of the exception
+     * @property exceptionMessage the message of the exception
+     * @property causee exception that is caused by this
+     * @property explanation the tag explaining the exception
+     * @property causeClass the class in which the exception is caused
+     * @property causeLine the line in which the exception is caused
+     * @property exceptionDoc a URL to the exceptions javadoc
+     * @property sealed whether this is the final exception data or not
+     */
+    data class ExceptionAnswer(
+        val exceptionName: String,
+        val exceptionMessage: String?,
+        val causee: String?,
```

Comment:
Typo

Reply:
the causee is the exception that is caused by the exception and not the exception that causes the causee
similar to the damaged entity being called damagee here:https://hub.spigotmc.org/javadocs/spigot/org/bukkit/event/entity/EntityDamageEvent.html#%3Cinit%3E(org.bukkit.entity.Entity,org.bukkit.event.entity.EntityDamageEvent.DamageCause,double)

Reply:
Can you also give proper answers or is that no longer cool and you have to react with emojis?

Reply:
I am too ashamed to give a proper answer... Good to know